src/dlr_tvm.cc

```diff
@@ -214,3 +223,32 @@ void TVMModel::UseCPUAffinity(bool use) {
     LOG(INFO) << "CPU Affinity is disabled";
   }
 }
+
+bool TVMModel::HasMetadata() const {
+  return this->metadata != nullptr;
+}
+
+const char* TVMModel::GetOutputName(const int index) const {
+  std::string name_str = this->metadata["Model"]["Outputs"][index]["name"];
+  return name_str.c_str();
+}
+
+int TVMModel::GetOutputIndex(const char* name) const {
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    std::string name_str = this->metadata["Model"]["Outputs"][i]["name"];
```

Comment:
Lets add `vector<string> output_names_` to `TVMModel` and fill it with output names from metadata in the constructor. Then we can use `output_names_` vector instead of using `metadata` each time users call `GetOutputIndex`.

Reply:
There is no benefit of storing it in two places. There is no drop in efficiency from the looks of it.